static/app/views/settings/account/notifications/notificationSettingsV2.tsx

```diff
@@ -0,0 +1,129 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+
+import AlertLink from 'sentry/components/alertLink';
+import {LinkButton} from 'sentry/components/button';
+import DeprecatedAsyncComponent from 'sentry/components/deprecatedAsyncComponent';
+import Form from 'sentry/components/forms/form';
+import JsonForm from 'sentry/components/forms/jsonForm';
+import {FieldObject} from 'sentry/components/forms/types';
+import Panel from 'sentry/components/panels/panel';
+import PanelBody from 'sentry/components/panels/panelBody';
+import PanelHeader from 'sentry/components/panels/panelHeader';
+import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
+import {IconMail, IconSettings} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {Organization} from 'sentry/types';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import withOrganizations from 'sentry/utils/withOrganizations';
+import {
+  NOTIFICATION_FEATURE_MAP,
+  NOTIFICATION_SETTINGS_PATHNAMES,
+  NOTIFICATION_SETTINGS_TYPES,
+  SELF_NOTIFICATION_SETTINGS_TYPES,
+} from 'sentry/views/settings/account/notifications/constants';
+import {NOTIFICATION_SETTING_FIELDS} from 'sentry/views/settings/account/notifications/fields2';
+import SettingsPageHeader from 'sentry/views/settings/components/settingsPageHeader';
+import TextBlock from 'sentry/views/settings/components/text/textBlock';
+
+type Props = DeprecatedAsyncComponent['props'] & {
+  organizations: Organization[];
+};
+
+function NotificationSettings({organizations}: Props) {
+  const checkFeatureFlag = (flag: string) => {
+    return organizations.some(org => org.features?.includes(flag));
+  };
+  const notificationFields = NOTIFICATION_SETTINGS_TYPES.filter(type => {
+    const notificationFlag = NOTIFICATION_FEATURE_MAP[type];
+    if (notificationFlag) {
+      return checkFeatureFlag(notificationFlag);
+    }
+    return true;
+  });
+
+  const renderOneSetting = (type: string) => {
+    const field = NOTIFICATION_SETTING_FIELDS[type];
+    return (
+      <FieldWrapper key={type}>
+        <div>
+          <FieldLabel>{field.label}</FieldLabel>
+          <FieldHelp>{field.help}</FieldHelp>
+        </div>
+        <IconWrapper>
+          <LinkButton
+            icon={<IconSettings size="sm" />}
+            size="sm"
+            borderless
+            aria-label={t('Notification Settings')}
+            to={`/settings/account/notifications/${NOTIFICATION_SETTINGS_PATHNAMES[type]}/`}
+          />
+        </IconWrapper>
+      </FieldWrapper>
+    );
+  };
+
+  const legacyFields = SELF_NOTIFICATION_SETTINGS_TYPES.map(
+    type => NOTIFICATION_SETTING_FIELDS[type] as FieldObject
+  );
+
+  // use 0 as stale time because we change the values elsewhere
+  const {data: initialLegacyData, isLoading} = useApiQuery<{[key: string]: string}>(
```

Comment:
you'll need to handle errors somehow

Reply:
@scttcper not sure what we'd do in an error scenario, any suggestions?